src/zope/i18n/compile.py

```diff
@@ -1,54 +1,61 @@
-from contextlib import closing
+
 import logging
-import os
-from os.path import join
-from stat import ST_MTIME
+logger = logging.getLogger('zope.i18n')
+
+def _cannot_compile_mo_file(domain, lc_messages_path):
+    logger.critical("Unable to compile messages: Python `gettext` library missing.")
+    return
 
-HAS_PYTHON_GETTEXT = True
 try:
     from pythongettext.msgfmt import Msgfmt
     from pythongettext.msgfmt import PoSyntaxError
-except ImportError:
+except ImportError: # pragma: no cover
     HAS_PYTHON_GETTEXT = False
+    compile_mo_file = _cannot_compile_mo_file
+else:
+    from contextlib import closing
+    import os.path
+    from os.path import join
```

Comment:
python-gettext is a 20 KB Python library with no external dependencies and with Python 3 support.  I think we can depend on it and avoid the conditional checks.

(I'm not a fan of how all the code in this module was indented in this PR: this makes the diff hard to read.  I had to switch to the Split view in GitHub and then switch between `?w=1`, which shows the changes but doesn't let me comment, and no `?w=1`, which lets me comment, but where I have to compare the functions by eyeball.)

Reply:
Dependency added.

Sorry, that was a big diff for that file.